update/application.go

```diff
@@ -0,0 +1,105 @@
+package update
+
+import (
+	"context"
+	"fmt"
+
+	apps "github.com/ninech/apis/apps/v1alpha1"
+	"github.com/ninech/nctl/api"
+	"github.com/ninech/nctl/api/util"
+	"github.com/ninech/nctl/internal/format"
+)
+
+// all fields need to be pointers so we can detect if they have been set by
+// the user.
+type applicationCmd struct {
+	Name     *string            `arg:"" help:"Name of the application."`
+	Git      *gitConfig         `embed:"" prefix:"git-"`
+	Size     *string            `help:"Size of the app."`
+	Port     *int32             `help:"Port the app is listening on."`
+	Replicas *int32             `help:"Amount of replicas of the running app."`
+	Hosts    *[]string          `help:"Host names where the application can be accessed. If empty, the application will just be accessible on a generated host name on the deploio.app domain."`
+	Env      *map[string]string `help:"Environment variables which are passed to the app at runtime."`
```

Comment:
we do not need pointer for slices and maps, right? 

> Map types are reference types, like pointers or slices...

https://go.dev/blog/maps#declaration-and-initialization

Reply:
But what if you want to empty the env with an update call? We have to able to tell if the user specified the flag `env=""` (meaning the update call should send that to the server) or the flag was not specified (Env == nil), meaning we should not touch the env.

Reply:
Sadly that is not how kong behaves as it will initialize the map for you. Given these flags:
```golang
Env      map[string]string
EnvPtr   *map[string]string

// cmd code
fmt.Println(cmd.Env)
fmt.Println(cmd.EnvPtr)
```

```
# without setting any flag
$ go run .
map[]
<nil>

# same when setting --env
$ go run . --env=""
map[]
<nil>

# when setting --env-ptr
$ go run . --env-ptr=""
map[]
&map[]
```

Reply:
then my bad. Sorry. I did not know that. This is a little different behavior than what you'd expect from Go.

Do you think that we could cover this case (empty Env vs nil) in our tests? Or Is it too much?